### PR TITLE
Feature: add recurring donation goal types

### PR DESCRIPTION
**Pull request by jonwaldstein**
## Description

This updates the goal options to include 3 new recurring specific goals:
1. Number of Subscriptions
2. Number of Subscribers
3. Subscription Amount Raised

Also included:
1. v2 to v3 form migration step updated
2. backwards compatibility meta updated

## Affects

- v3 Form goal
- v2 to v3 form migration

## Visuals

Screenshot 2023-10-03 at 10 39 26 AM

Screenshot 2023-10-03 at 10 38 20 AM

## Testing Instructions

- Create a v3 form with recurring enabled
- Use the new goal options
- Create a mix of one-time and recurring donations to test out the goals

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205643550365341
  - https://app.asana.com/0/0/1205586807266064

### CODE REVIEW

**Comment by jonwaldstein** [comment]
@kjohnson this is ready for review.

File: src/DonationForms/Properties/FormSettings.php [hunk 1]
```diff
@@ -8,6 +8,8 @@
 use Give\Framework\Support\Contracts\Arrayable;
 use Give\Framework\Support\Contracts\Jsonable;
 
+use const FILTER_VALIDATE_BOOLEAN;
```
[the remark this answers is not in the document]
Comment (jonwaldstein) [reply src/DonationForms/Properties/FormSettings.php]: resolved b44b4d1a

File: src/DonationForms/DataTransferObjects/DonationFormGoalData.php [hunk 2]
```diff
@@ -61,7 +67,39 @@ public function getCurrentAmount(): int
...
+
+        if ($this->goalType->isAmountFromSubscriptions()) {
+            return $donationFormRepository->getTotalInitialAmountFromSubscriptions($this->formId);
+        }
+
+        if ($this->goalType->isDonorsFromSubscriptions()) {
+            return $donationFormRepository->getTotalNumberOfDonorsFromSubscriptions($this->formId);
+        }
+
+        return null;
```
[the remark this answers is not in the document]
Comment (jonwaldstein) [reply src/DonationForms/DataTransferObjects/DonationFormGoalData.php:82]: hmm yeah good point, I wonder if having a default goalType is even a good thing.  I would think amount is the most common goal type anyway so maybe for consistency i'll make that the default here and we can circle back to removing the defaults if need be.

[the remark this answers is not in the document]
Comment (jonwaldstein) [reply src/DonationForms/DataTransferObjects/DonationFormGoalData.php]: Resolved [6742577](https://github.com/impress-org/givewp/pull/6976/commits/67425774d0d9033cfdad53c3e70fa28d7167271c)

**Review by kjohnson — APPROVED** [review]

**Review by rickalday — APPROVED** [review]
Passed manual QA tests

**Pull request merged**